pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -754,7 +772,13 @@ func (m *kubeGenericRuntimeManager) SyncPod(pod *v1.Pod, podStatus *kubecontaine
 			klog.V(4).Infof("Determined the ip %v for pod %q after sandbox changed", podIPs, format.Pod(pod))
 		}
 	}
+	return
+}
 
+//  5. Create ephemeral containers.
+//  6. Create init containers.
+//  7. Create normal containers.
+func (m *kubeGenericRuntimeManager) SyncPod(pod *v1.Pod, podStatus *kubecontainer.PodStatus, pullSecrets []v1.Secret, backOff *flowcontrol.Backoff, podIPs []string, podSandboxID string) (result kubecontainer.PodSyncResult) {
```

Comment:
podSandboxID can be accessed via `podContainerChanges.SandboxID`, and doesn't need to be passed in as an argument.

Reply:
If I use podContainerChanges.SandboxID as podSandboxID at the beginning of SyncPodContainers(), the following test would fail:
```
    TestSyncPodWithInitContainers: kuberuntime_manager_test.go:630: 
        	Error Trace:	kuberuntime_manager_test.go:630
        	Error:      	Received unexpected error:
        	            	container (name, attempt, sandboxID)=("init1", 0, "foo_new_12345678_0") not found
        	Test:       	TestSyncPodWithInitContainers
```

Reply:
You may have to update the unit test as well.